tests/avro/test_resolver.py

```diff
@@ -372,7 +372,7 @@ def test_writer_ordering() -> None:
         ),
     )
 
-    expected = StructWriter(((1, DoubleWriter()), (0, StringWriter())))
+    expected = StructWriter(((0, DoubleWriter()), (1, StringWriter())))
```

Comment:
Not sure if this change is semantically correct. This test is affected because `resolve_writer` compares the two given schemas (`record_schema` and `file_schema`)

https://github.com/apache/iceberg-python/blob/7bd5d9e6c32bcc5b46993d6bfaeed50471e972ae/pyiceberg/avro/resolver.py#L200-L214

Previously, comparison returned `False` due to different ordering